R/construct_design.R

```diff
@@ -82,56 +82,56 @@
   # 1. lhs is a step
   # 2. lhs is a design
   
-  if (is_missing(rhs) || is_null(rhs)) {
-    
+  if (missing(rhs)) {
+    rhs <- NULL
     qs <- enquos(lhs)
-    
   } else {
+    qs <- enquos(lhs, rhs)
+  }
+  
+  if (!inherits(rhs, "dd") && !inherits(rhs, "function") && !is.null(rhs)) {
+    stop("The right hand side does not appear to be a DeclareDesign object or a function", call. = FALSE)
+  }
+  
+  lhs <- if (inherits(lhs, "design")) {
+      Filter(Negate(is_autogenerated), lhs)
+    } else wrap_step(lhs, f_rhs(qs[[1]]))
+
+  rhs <- if (inherits(rhs, "design")) {
+    Filter(Negate(is_autogenerated), rhs)
+  } else if(!is.null(rhs)) wrap_step(rhs, f_rhs(qs[[2]]))
+
+  to_rename <- names(lhs) %i% names(rhs)
+  
+  for(i in match(to_rename, names(rhs))) {
+    if(attr(rhs, "valid")) next;
+    new_nm <- names(rhs)[i]
     
-    if (!inherits(rhs, "dd") && !inherits(rhs, "function")) {
-      stop("The right hand side does not appear to be a DeclareDesign object or a function", call. = FALSE)
-    }
+    while(new_nm %in% union(names(lhs), names(rhs)))
+      new_nm <- paste0(new_nm, "._")
```

Comment:
Needs to add numbers like data.frame does.

Reply:
great, just made commits along these lines